utils/ws_client.py

```diff
@@ -32,18 +30,26 @@ def __init__(self, url: str, timeout: int = 2):
         self.logger = logging.getLogger()
         self.logger.setLevel(logging.INFO)
 
-    def open(self, keep_alive: bool = True) -> None:
+    def open(self, keep_alive: bool = True, origin: dict =None) -> None:
         ''' Open WebSocket connection.
                 Args:
                     keep_alive(bool): Bool which states if connection should be kept, by default sets to `True`.
+                    origin (dict): Specifies origin while creating websocket connection.
         '''
         try:
-            self.websocket = websocket.create_connection(
-                self.url,
-                self.timeout,
-                origin=parse_url_and_return_origin(self.url),
-                sslopt={'cert_reqs': ssl.CERT_NONE},
-                enable_multithread=True)
+            if origin is not None:
+                self.websocket = websocket.create_connection(
+                        self.url,
+                        self.timeout,
+                        origin=origin,
+                        sslopt={'cert_reqs': ssl.CERT_NONE},
+                        enable_multithread=True)
+            else:
+                self.websocket = websocket.create_connection(
+                        self.url,
+                        self.timeout,
+                        sslopt={'cert_reqs': ssl.CERT_NONE},
+                        enable_multithread=True)
```

Comment:
Avoid negative conditionals
```suggestion
            if origin:
                self.websocket = websocket.create_connection(
                        self.url,
                        self.timeout,
                        sslopt={'cert_reqs': ssl.CERT_NONE},
                        enable_multithread=True)     
            else:
                self.websocket = websocket.create_connection(
                        self.url,
                        self.timeout,
                        origin=origin,
                        sslopt={'cert_reqs': ssl.CERT_NONE},
                        enable_multithread=True)
```

Reply:
👍